src/main/java/Duke.java

```diff
@@ -30,6 +38,21 @@ public Duke() {
 
     private static List<Task> tasks = new ArrayList<>();
 
+    /**
+     * Parses the input entered by the client.
+     * The following are valid commands that Duke can process:
+     * list - lists all the tasks that are stored.
+     * done [index] - marks the task of a particular index as done.
+     * delete [index] - deletes the task of a particular index.
+     * todo [description] - adds the Todo task to the list.
+     * deadline [description] /by [date] [time] - adds the Deadline task to the list.
+     * event [description] /at [date] [time] - adds the Event task to the list.
+     * If an exit command is entered, it is processed,
+     * then the goodbye message is printed and the program exits from the loop.
+     *
+     * @throws DukeException If the command is invalid or the task enquired doesn't exists.
+     * @throws DateTimeParseException If the date of the deadline or event is not formatted properly.
+     */
```

Comment:
Commendable effort to have a segment that is clear as it can easily be referenced when lost in the code.